packages/service-library/src/servicelib/redis.py

```diff
@@ -223,17 +223,20 @@ class RedisClientsManager:
     databases: set[RedisDatabase]
     settings: RedisSettings
 
-    _client_sdks: dict[RedisDatabase, RedisClientSDK] = field(default_factory=dict)
+    _client_sdks: dict[RedisDatabase, RedisClientSDKHealthChecked] = field(
+        default_factory=dict
+    )
 
     async def setup(self) -> None:
         for db in self.databases:
-            self._client_sdks[db] = client_sdk = RedisClientSDK(
+            self._client_sdks[db] = RedisClientSDKHealthChecked(
```

Comment:
I noticed that`RedisClientSDKHealthChecked` introduces a task upon setup that periodically pings redis to evaluate its health. Here you are creating a different client per database, meaning that you are adding as much amount of background tasks to check exactly the same. 
Some questions:
Q1: why to check the **same thing** so many times?
Q2: On a single client: why do you need a background task to check the health? Wouldn't it suffice checking when you call `is_healthy()` member function?

Reply:
Q1: my reason for checking every connection is the following: nothing prevents for only 1 connection out of all of them to be degraded. I prefer safety over the low overhead of this check.

Q2: the background task avoids a vey long wait if something is now working as expected, or if system performance is degraded. The reason being I don't want to block waiting in real time for the check, but I want to read it from a field and then decide what to do immediately. This makes the health check endpoints and everything relating on this faster

Q3: I'll have a look into this

Reply:
Q3: if one of the setup fails the entire boot of the application will fail so I think it's OK like this.

Reply:
I've replied to these questions in the feedback below